util/types/datum.go

```diff
@@ -315,8 +315,10 @@ func (d *Datum) SetValue(val interface{}) {
 		d.SetBytes(x)
 	case mysql.Bit:
 		d.SetMysqlBit(x)
-	case mysql.Decimal:
+	case *mysql.Decimal:
 		d.SetMysqlDecimal(x)
+	case mysql.Decimal:
```

Comment:
We still have some setValue statements to put decimal in datum. But this case will be removed in the futher.